### PR TITLE
Possibility to save all map layers at the same time

### LINKED ISSUE #4016: Possibility to save all map layers at the same time and not one per one
Right now, in frontend, we have to save layers attributes sequentially, waiting for the response until we save the next layer in the array. It'd would be great if we could save all layers in only one call :angel:.

Ticket coming from https://github.com/CartoDB/cartodb/pull/3990.

### PR DESCRIPTION
**Pull request by xavijam**
Basically we have a problem when we need to save the state of all layers those belong to a visualization.
We used to sent as many PUTs as layers we have in the visualization. From now on, we want to just make a request with the needed info.

Fixes #4016.
@juanignaciosl

### CODE REVIEW

**Comment by xavijam** [comment]
So, with the current changes the request works correctly but:

- Map is not refreshed.
- Layers order is not being render properly, mainly because:
https://github.com/CartoDB/cartodb/blob/4016-Possibility_to_save_all_map_layers_at_the_same_time/lib/assets/javascripts/cartodb/table/layer_panel_view.js#L356
Seems like we need to sort layers or change that part of code and get 'order' attribute.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4659/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4660/))

**Comment by alonsogarciapablo** [comment]
Hey! What's the status of this PR? @xavijam are you planning to do some more work here? (I'm asking because of the two caveats that you mentioned in your comment). This is somehow related to [this PR](https://github.com/CartoDB/cartodb/pull/4286), so it'd be nice to QA everything at the same time.

File: lib/assets/javascripts/cartodb/models/map.js [hunk 1]
```diff
@@ -433,39 +433,25 @@ cdb.admin.Layers = cdb.geo.Layers.extend({
...
-              opts && opts.complete && opts.complete();
-            }
-          },
-          error: function() {
-            opts && opts.error && opts.error();
-            opts && opts.complete && opts.complete();
-          }
-        }
-      );
+    this.save(null, opts);
```
Comment (alonsogarciapablo) [inline lib/assets/javascripts/cartodb/models/map.js]: Since `save` is also a one-liner, we could move its code here instead of adding a new public method.

Comment (xavijam) [inline lib/assets/javascripts/cartodb/models/map.js]: Totally agree, but we still have the problem with the "invalidation" of the tiles when save is completed.

**Comment by ztephm** [comment]
Hi, this issue is happening still for S/B 5831712

Think you guys already got this part, but:

Expected: 
- drag top layer to bottom of layer stack in Map View
- data layers in map re-order according to new layer order

Happening:
- drag top layer to bottom of layer stack in Map View
- data layers are not re-ordering

Replicated here:
1. Add line layer
2. Add point layer
3. Add polygon layer
4. Drag polygon layer to bottom

Polygon layer shows at bottom of Map View layer stack but polygons are still on top of other line and point layers:
http://cdb.io/1I3nhYG

screen shot 2015-07-15 at 12 36 31 pm

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5097/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5100/))

**Comment by alonsogarciapablo** [comment]
I think this is now ready and working! :relieved:

@xavijam Could you PTAL? Thanks!

**Comment by xavijam** [comment]
Looks great man :)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5103/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5139/))

**Comment by xavijam** [comment]
\o/ :dancer: :grapes:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5142/))

**Comment by alonsogarciapablo** [comment]
Deployed and working :+1:. Great job guys!

**Comment by juanignaciosl** [comment]
_Yahoo!_ :-)

**Pull request merged**